src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationOptions.cs

```diff
@@ -49,17 +49,15 @@ public string DisplayName
         }
 
         /// <summary>
-        /// Defines whether access and refresh tokens should be stored in the
-        /// <see cref="ClaimsPrincipal"/> after a successful authorization with the remote provider.
-        /// This property is set to <c>false</c> by default to reduce
-        /// the size of the final authentication cookie.
+        /// Gets or sets the time limit for completing the authentication flow (15 minutes by default).
         /// </summary>
-        public bool SaveTokensAsClaims { get; set; }
+        public TimeSpan RemoteAuthenticationTimeout { get; set; } = TimeSpan.FromMinutes(15);
 
         /// <summary>
-        /// Gets or sets the time limit for completing the authentication flow (15 minutes by default).
+        /// An optional location to store access and refresh tokens after a successful authorization with the remote provider.
+        /// This can also be provided via dependency injection.
         /// </summary>
-        public TimeSpan RemoteAuthenticationTimeout { get; set; } = TimeSpan.FromMinutes(15);
+        public ITokenStore TokenStore { get; set; }
```

Comment:
You know, there's nothing specific to Remote here, we could move this into the base AuthenticationOptions, and then consolidate the logic in the base AuthenticationMiddleware so every ctor doesn't have to set it.  This would let any auth middleware put things into the TokenStore as well if they wanted to.  It doesn't hurt, and it cleans up the code a bit too...